### PR TITLE
Add the support of str to-int to the into int command

### PR DESCRIPTION
**Pull request by onthebridgetonowhere**
@fdncred, @kubouch 
this PR adds the `str to-int` support for the `into int` command. Let me know if you have any feedback.

### CODE REVIEW

**Comment by fdncred** [comment]
nice! so `into int` now does eveything it did before plus everything `str to-int` did?

File: crates/nu-command/src/conversions/into/int.rs [hunk 1]
```diff
@@ -134,28 +181,120 @@ pub fn action(input: &Value, span: Span) -> Value {
...
+        let expected = Value::test_int(10);
+
+        let actual = action(&word, Span::unknown(), 10);
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn turns_binary_to_integer() {
+        let s = Value::test_string("0b101");
+        let actual = action(&s, Span::unknown(), 10);
```
Comment (jntrnr) [inline crates/nu-command/src/conversions/into/int.rs:281]: I'm surprised this doesn't have to be a 2 instead of a 10. Do you know why it works, by chance?

Comment (onthebridgetonowhere) [reply crates/nu-command/src/conversions/into/int.rs:281]: Good catch. When a string is matched, we check if the string starts with `0b`. If yes, then the "given" radix is ignored and 2 is used instead. Same thing for a string starting with "0x", we default to 16. You can see this in line 188 and line 216. 

I think the code can be slightly improved.

**Comment by onthebridgetonowhere** [comment]
@fdncred yes. I added all the previous test cases of `str to-int` to `into int`, so it should be complete now.

**Comment by jntrnr** [comment]
thanks! We can always improve things but I think this is good

**Pull request merged**